src/Monomer/Graphics/FFI.chs

```diff
@@ -118,6 +119,23 @@ withText t = useAsCString (T.encodeUtf8 t)
 withNull :: (Ptr a -> b) -> b
 withNull f = f nullPtr
 
+allocCUString :: ByteString -> ((Ptr CUChar, CInt) -> IO a) -> IO a
```

Comment:
@klausweiss I'm thinking maybe something like this could work (I have not tested it, and it may not compile as it stands):

```haskell
withByteString :: ByteString -> (CString -> IO b) -> IO b
withByteString bs = useAsCString bs

{# fun unsafe fmCreateFontMem {`FMContext', withCString*`Text', withByteString*`ByteString'&} -> `Int' #}
```

Reply:
Unfortunately the C API requires it to use `Ptr CUChar` instead of `Ptr CChar`, which `CString` uses. I've made some progress (see [this comment](https://github.com/fjvallarino/monomer/pull/199#issuecomment-1206556611)), but that's still not it.